File: api/src/main/java/com/api/app/entity/OrderDetail.java
package com.api.app.entity;

import io.swagger.v3.oas.annotations.media.Schema;
import lombok.Getter;
import lombok.Setter;
import org.apache.ibatis.type.Alias;

import java.io.Serializable;
import java.time.LocalDateTime;

/**
 * @author system
 * @version 1.0
 * @since 2025-10-27
 */
@Alias("OrderDetail")
@Getter
@Setter
public class OrderDetail implements Serializable {
    private static final long serialVersionUID = 3456789012345678901L;

    @Schema(description = "주문번호")
    private String orderNo;

    @Schema(description = "주문순번")
    private Long orderSequence;

    @Schema(description = "주문처리순번")
    private Long orderProcessSequence;

    @Schema(description = "상위주문처리순번")
    private Long upperOrderProcessSequence;

    @Schema(description = "클레임번호")
    private String claimNo;

    @Schema(description = "상품번호")
    private String goodsNo;

    @Schema(description = "단품번호")
    private String itemNo;

    @Schema(description = "수량")
    private Long quantity;

    @Schema(description = "주문상태코드")
    private String orderStatusCode;

    @Schema(description = "배송구분코드")
    private String deliveryTypeCode;

    @Schema(description = "주문유형코드")
    private String orderTypeCode;

    @Schema(description = "주문접수일시")
    private LocalDateTime orderAcceptDtm;

    @Schema(description = "주문완료일시")
    private LocalDateTime orderFinishDtm;
}

File: api/src/main/kotlin/com/api/app/entity/OrderDetail.kt
package com.api.app.entity

import jakarta.persistence.*
import java.io.Serializable
import java.time.LocalDateTime

@Embeddable
data class OrderDetailId(
    @Column(name = "ORDER_NO", length = 15) val orderNo: String = "",
    @Column(name = "ORDER_SEQUENCE", columnDefinition = "numeric") val orderSequence: Long = 0,
    @Column(name = "ORDER_PROCESS_SEQUENCE", columnDefinition = "numeric") val orderProcessSequence: Long = 0
) : Serializable

@Entity
@Table(name = "ORDER_DETAIL")
class OrderDetail : SystemEntity() {

    @EmbeddedId
    var id: OrderDetailId = OrderDetailId()

    @Column(name = "UPPER_ORDER_PROCESS_SEQUENCE", columnDefinition = "numeric")
    var upperOrderProcessSequence: Long? = null

    @Column(name = "CLAIM_NO", length = 15)
    var claimNo: String? = null

    @Column(name = "GOODS_NO", nullable = false, length = 15)
    var goodsNo: String = ""

    @Column(name = "ITEM_NO", nullable = false, length = 3)
    var itemNo: String = ""

    @Column(name = "QUANTITY", nullable = false, columnDefinition = "numeric")
    var quantity: Long = 0

    @Column(name = "ORDER_STATUS_CODE", nullable = false, length = 3)
    var orderStatusCode: String = ""

    @Column(name = "DELIVERY_TYPE_CODE", nullable = false, length = 3)
    var deliveryTypeCode: String = ""

    @Column(name = "ORDER_TYPE_CODE", nullable = false, length = 3)
    var orderTypeCode: String = ""

    @Column(name = "ORDER_ACCEPT_DTM")
    var orderAcceptDtm: LocalDateTime? = null

    @Column(name = "ORDER_FINISH_DTM")
    var orderFinishDtm: LocalDateTime? = null

    // Convenience accessors
    val orderNo: String get() = id.orderNo
    val orderSequence: Long get() = id.orderSequence
    val orderProcessSequence: Long get() = id.orderProcessSequence
}
